src/parachain/loans.ts

```diff
@@ -435,18 +540,29 @@ export class DefaultLoansAPI implements LoansAPI {
         return newMonetaryAmount(amount, rewardCurrency);
     }
 
+    async _getExchangeRate(fromCurrency: CurrencyExt): Promise<ExchangeRate<WrappedCurrency, CurrencyExt>> {
+        const wrappedCurrencyId = this.api.consts.escrowRewards.getWrappedCurrencyId;
+        const wrappedCurrency = tokenSymbolToCurrency(wrappedCurrencyId.asToken);
+        if (isCurrencyEqual(fromCurrency, wrappedCurrency)) {
+            const wrappedCurrencyToBitcoinRate = Big(1); // Use exchange rate of 1 between wrapped BTC and real BTC.
+            return new ExchangeRate(wrappedCurrency, wrappedCurrency, wrappedCurrencyToBitcoinRate);
+        }
+        return this.oracleAPI.getExchangeRate(fromCurrency);
+    }
+
     async _getLoanAsset(
         underlyingCurrencyId: InterbtcPrimitivesCurrencyId,
         marketData: LoansMarket
     ): Promise<[CurrencyExt, LoanAsset]> {
         const underlyingCurrency = await currencyIdToMonetaryCurrency(this.api, underlyingCurrencyId);
 
-        const [lendApy, borrowApy, [totalLiquidity, availableCapacity, totalBorrows], rewardCurrency] =
+        const [lendApy, borrowApy, [totalLiquidity, availableCapacity, totalBorrows], rewardCurrency, exchangeRate] =
             await Promise.all([
                 this._getLendApy(underlyingCurrencyId),
                 this._getBorrowApy(underlyingCurrencyId),
                 this._getTotalLiquidityCapacityAndBorrows(underlyingCurrency, underlyingCurrencyId),
                 this._getRewardCurrency(),
+                this._getExchangeRate(underlyingCurrency),
```

Comment:
Will this become the only place where oracle prices are fetched in the UI?

Reply:
No, the oracle price is used fetched in other places in lib that are used in UI too - vault capacity calculation, vault apy calculation.

Reply:
For now it's fine but the UI could optimize this by fetching exchange rates just once and then passing them down to the various components, the same way `loanAssets` is passed in `getLoanCollateralInfo`. If you think this makes sense you can create a GitHub issue in the UI/lib.

Reply:
I think it's not necessary at the moment since in the other place the exchange rate is fetched only for one currency - so fetching all extra exchange rates for loan currencies would not make the UI more efficient.